string filepath
---
int32 objectIndex
